platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/configuration/problems/DecoratedReportProblem.kt

```diff
@@ -25,11 +27,124 @@ import org.gradle.internal.problems.failure.StackTraceRelevance
 data class DecoratedReportProblem(
     val trace: PropertyTrace,
     val message: StructuredMessage,
-    val failure: DecoratedFailure? = null,
-    val documentationSection: DocumentationSection? = null
+    val failure: DecoratedFailure?,
+    val docLink: String?,
+    val kind: String
 )
 
 
+final class DecoratedReportProblemJsonSource(val problem: DecoratedReportProblem) : JsonSource {
```

Comment:
💅  Kotlin classes are final by default

Also, does the `problem` need to be public?